src/MLJTreeParzen.jl

```diff
@@ -242,24 +245,35 @@ function MLJTuning.models!(
 
     # we only ever do this when num_hist == 0
     for i in 1:num_suggest
-        push!(candidates, ask(state.suggestions[i]))
+        push!(candidates, ask(space.suggestions[i]))
     end
 
     # when num_hist == 0 this is the full number of random (independent draws) (minus suggestions as drawn above)
     # else its the max number of times we can ask TP from its distributions
     for i in 1:max_draws
-        push!(candidates, ask(state.space, trialhist, strategy.config))
+        push!(candidates, ask(space.space, trialhist, strategy.config))
     end
 
-    return [
+    newstate = (space=space, trialhist=trialhist)
```

Comment:
I spent a while (I mean just a few minutes) trying to grok why we could eat things from the state in the way we were, but I see now with this more functional approach to `models` we return a constructed state: I approve